paper/paper.Rmd

```diff
@@ -0,0 +1,149 @@
+---
+title: "Static Code Analysis for R"
+date: "`r Sys.Date()`"
+tags: ["R", "linter", "tidyverse"]
+
+authors:
+  - name: Jim Hester
+    affiliation: 1
+    orcid: ~
+
+affiliations:
+  - index: 1
+    name: Netflix
+
+output: 
+  rticles::joss_article:
+    keep_tex: true
+bibliography: paper.bib
+csl: apa.csl
+journal: JOSS
+link-citations: yes
+---
+
+```{r, warning=FALSE, message=FALSE, echo=FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  out.width = "100%",
+  dpi = 300,
+  comment = "#>"
+)
+
+options(width = 60)
+```
+
+# Statement of Need
+
+The R programming language [@base2023] is a popular choice for statistical analysis and visualization, and is used by a wide range of researchers and data scientists. The `{lintr}` package is an open-source R package that provides static code analysis to check for a variety of common problems related to readability, efficiency, consistency, style, etc. It is designed to be easy to use and integrate into existing workflows, and can be run from the command line or used as part of an automated build or continuous integration process. `{lintr}` also integrates with a number of popular IDEs and text editors, such as RStudio and Visual Studio Code, making it convenient for users to run `{lintr}` checks on their code as they work.
+
+# Features
+
+There are over 85 linters offered by `{lintr}`!
+
+```{r}
+library(lintr)
+
+length(all_linters())
+```
+
+Naturally, we can't discuss all of them here. To see details about all available linters, we encourage readers to see <https://lintr.r-lib.org/dev/reference/index.html#individual-linters>.
+
+We will showcase one linter for each kind of common problem found in R code.
+
+- **Best practices**
+
+`{lintr}` offers linters that can detect problematic antipatterns and suggest alternative patterns that follow best practices.
+
+For example, usage of vectorized `&` and `|` logical operators in conditional statements is error-prone, and scalar `&&` and `||`, respectively, are to be preferred. The `vector_logic_linter()` linter detects such problematic usages.
```

Comment:
If the audience may not know R, it will be good to discuss the difference of vector/scalar operators a bit more in the context of R as a language.

Reply:
Hmm, I am not sure if we need to get into that. 

Will it be better to use some other linter as an example to avoid having to explain vector/scalar operators?